delta/service/src/test/scala/ch/epfl/bluebrain/nexus/delta/service/resources/ResourceEventExchangeSpec.scala

```diff
@@ -106,13 +106,24 @@ class ResourceEventExchangeSpec
       result.value.source shouldEqual source
       result.value.resource shouldEqual resRev2
       result.metadata.value shouldEqual ()
+      result.tag shouldEqual None
     }
 
-    "return the latest resource state from the event at a particular tag" in {
+    "return the resource state from the event at a particular tag" in {
       val result = exchange.toResource(deprecatedEvent, Some(tag)).accepted.value
       result.value.source shouldEqual source
       result.value.resource shouldEqual resRev1
       result.metadata.value shouldEqual ()
+      result.tag.value shouldEqual tag
+    }
+
+    "return the latest resource state if the tag has been deleted" in {
```

Comment:
Should this fail if the tag was not found?

Reply:
The way an `EventExchange` needs to work now:
if the tag is found return resource at the tag with `Some(tag)`,
if the tag is not found, return latest version of resource with `None`
It's not the most efficient way, but we cannot filter out events anymore based on tag, because we need to delete resources from the index.